splunk_eventgen/lib/generatorplugin.py

```diff
@@ -206,16 +165,63 @@ def setupBackfill(self):
                 except Exception as ex:
                     self.logger.error("Failed to parse end '%s' for sample '%s', treating as number of executions" % (s.end, s.name))
                     raise
+
     def run(self, output_counter=None):
         if output_counter is not None and hasattr(self.config, 'outputCounter') and self.config.outputCounter:
             # Use output_counter to calculate throughput
             self._out.setOutputCounter(output_counter)
         self.gen(count=self.count, earliest=self.start_time, latest=self.end_time, samplename=self._sample.name)
-        #TODO: Make this some how handle an output queue and support intervals and a master queue
+        # TODO: Make this some how handle an output queue and support intervals and a master queue
         # Just double check to see if there's something in queue to flush out at the end of run
         if len(self._out._queue) > 0:
             self.logger.debug("Queue is not empty, flush out at the end of each run")
             self._out.flush()
 
+    def replace_tokens(self, eventsDict, earliest, latest, ignore_tokens=False):
```

Comment:
Do we even need ignore_tokens?  What's the usecase for ignoring it?

Reply:
The generatorplugin calls replace_tokens every time it builds events, however the perDayVolume generator now calls replace_tokens before building events. It's preventing us from parsing the sample for tokens more than once, but the code should work fine without it because there will be no tokens left to replace anyways. I would prefer not to use it, if you think the extra parsing won't be costly.